### ISSUE TITLE
Unexpected generic exception in triple

### ISSUE THREAD

**Issue by focuse8** [apache/dubbo#13368]
When the second parameter of GenericService.$invoke is passed null, version 3.2.7 fills it with the parameter type of $invoke, causing the provider to not be able to find the corresponding method.
### Environment

* Dubbo version: 3.2.7
* Operating System version: xxx
* Java version: 1.8

### Steps to reproduce this issue

1. Set dubbo version 3.2.7
2. Use triple protocol generics genericService.$invoke("methodName", null, new Object[])

Pls. provide [GitHub address] to reproduce this issue.

### Expected Behavior
Return normally

### Actual Behavior
The provider cannot find the corresponding method

Cause Analysis:
Use org.apache.dubbo.rpc.protocol.tri.TripleInvoker#invokeUnary to obtain the actual parameter type using 
```java
org.apache.dubbo.rpc.support.RpcUtils#getParameterTypes
public static Class<?>[] getParameterTypes(Invocation invocation) {
         if ($INVOKE.equals(invocation.getMethodName())
             && invocation.getArguments() != null
             && invocation.getArguments().length > 1
             && invocation.getArguments()[1] instanceof String[]) {
             String[] types = (String[]) invocation.getArguments()[1];
             if (types == null) {
                 return new Class<?>[0];
             }
             Class<?>[] parameterTypes = new Class<?>[types.length];
             for (int i = 0; i < types.length; i++) {
                 parameterTypes[i] = ReflectUtils.forName(types[i]);
             }
             return parameterTypes;
         }
         return invocation.getParameterTypes();
     }
```
If null is passed, invocation.getArguments()[1] instanceof String[] will not be established, resulting in the actual parameter type returned being a generic parameter type [String, Ljava.lang.String, Ljava.lang.Object]

The request received by the provider is as shown below,
![image](https://github.com/apache/dubbo/assets/7105269/5f1e7dcd-60be-4eaa-864a-f8290b74a187)

---

### Environment

* Dubbo version: 3.2.7
* Operating System version: xxx
* Java version: 1.8

### Steps to reproduce this issue

1. 设置dubbo版本3.2.7
2. triple协议泛型掉用 genericService.$invoke("methodName", null, new Object[])

Pls. provide [GitHub address] to reproduce this issue.

### Expected Behavior
正常返回

### Actual Behavior
提供者找不到对应方法

原因分析：
org.apache.dubbo.rpc.protocol.tri.TripleInvoker#invokeUnary 里面使用org.apache.dubbo.rpc.support.RpcUtils#getParameterTypes获取实际参数类型
public static Class<?>[] getParameterTypes(Invocation invocation) {
        if ($INVOKE.equals(invocation.getMethodName())
            && invocation.getArguments() != null
            && invocation.getArguments().length > 1
            && invocation.getArguments()[1] instanceof String[]) {
            String[] types = (String[]) invocation.getArguments()[1];
            if (types == null) {
                return new Class<?>[0];
            }
            Class<?>[] parameterTypes = new Class<?>[types.length];
            for (int i = 0; i < types.length; i++) {
                parameterTypes[i] = ReflectUtils.forName(types[i]);
            }
            return parameterTypes;
        }
        return invocation.getParameterTypes();
    }
如果传的是null，则  invocation.getArguments()[1] instanceof String[]不会成立，导致返回的实际参数类型泛型的参数类型 [String, Ljava.lang.String, Ljava.lang.Object]

提供者收到的请求如下图，
![image](https://github.com/apache/dubbo/assets/7105269/5f1e7dcd-60be-4eaa-864a-f8290b74a187)

**Comment by AlbumenJ** [comment]
Can you please provider the error log including exception stack trace?

**Comment by focuse8** [comment]
> Can you please provider the error log including exception stack trace?
Acturally, our method named ExperimentDetailApi.queryForGatewayRoute has only one object param， but in generics there are three params which acturally are  the params of GenericService.$invoke.
 The stack is as below:

Invocation of init method failed; nested exception is org.apache.dubbo.rpc.StatusRpcException: UNKNOWN : com......ExperimentDetailApi.queryForGatewayRoute(java.lang.String, [Ljava.lang.String;, [Ljava.lang.Object;)
org.apache.dubbo.rpc.RpcException: com......ExperimentDetailApi.queryForGatewayRoute(java.lang.String, [Ljava.lang.String;, [Ljava.lang.Object;)
	at org.apache.dubbo.rpc.filter.GenericFilter.invoke(GenericFilter.java:188)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:326)
	at org.apache.dubbo.rpc.filter.ClassLoaderFilter.invoke(ClassLoaderFilter.java:54)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:326)
	at org.apache.dubbo.rpc.filter.EchoFilter.invoke(EchoFilter.java:41)

	at org.springframework.context.annotation.CommonAnnotationBeanPostProcessor.postProcessProperties(CommonAnnotationBeanPostProcessor.java:321)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1420)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:593)

**Comment by AlbumenJ** [comment]
Do you mean that you have defined a service which have the params like `java.lang.String, [Ljava.lang.String;, [Ljava.lang.Object;` or you have invoked a service by Dubbo generic service with `GenericService` interface?

**Comment by focuse8** [comment]
> Do you mean that you have defined a service which have the params like `java.lang.String, [Ljava.lang.String;, [Ljava.lang.Object;` or you have invoked a service by Dubbo generic service with `GenericService` interface?


I have invoked a service with ’GenericService‘ interface.   my service has just only one param,  that is java.lang.Object.  But when I invoked it using  GenericService.$invoke("queryForGatewayRoute", null,  args)， the provider threw an NoSuchMethodExceptio. Because  the provider received a request with three params `java.lang.String, [Ljava.lang.String;, [Ljava.lang.Object;`. As I debuged above,   the org.apache.dubbo.rpc.support.RpcUtils#getParameterTypes method fills up params with  `java.lang.String, [Ljava.lang.String;, [Ljava.lang.Object;` in dubbo 3.2.7.
